### PR TITLE
feat(git-hooks): adds git precommit hooks

### LINKED ISSUE #33: Add git pre-commit hooks to lint with prettier
I've set this up in a few projects and it's awesome - then you don't need to worry about folks running `yarn run fmt` before checking code in.

### PR DESCRIPTION
**Pull request by goldcaddy77**
Fixes #33

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "author": "Christian Joudrey",
   "main": "lib/index.js",
   "scripts": {
-    "fmt": "node_modules/prettier/bin/prettier.js --single-quote --trailing-comma es5 --write -l \"{test,src}/**/*.js\"",
+    "fmt": "prettier --single-quote --trailing-comma es5 --write -l \"{test,src}/**/*.js\"",
```
Comment (goldcaddy77) [inline package.json]: Not necessary as `yarn run` sets up the path and uses script in `/node_modules/.bin`

**Comment by goldcaddy77** [comment]
@cjoudrey - have you used the `husky` + `lint-staged` stack before?  This runs a git hook that auto-formats the files and re-git adds them before committing.

screen shot 2017-08-16 at 5 08 37 pm

**Comment by cjoudrey** [comment]
I haven't used it before. I believe we've chatted about it though. 😀 

I can't think of any reason why we wouldn't want this.

File: package.json (hunk 1 above)
Comment (cjoudrey) [inline package.json]: I wouldn't mind just 🔥 this since we now have the pre-commit hook.

**Comment by cjoudrey** [comment]
Thanks again 😁

**Pull request merged**